client/rest-high-level/src/test/java/org/elasticsearch/client/EqlIT.java

```diff
@@ -21,58 +21,145 @@
+import org.elasticsearch.action.bulk.BulkResponse;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.action.support.WriteRequest;
 import org.elasticsearch.client.eql.EqlSearchRequest;
 import org.elasticsearch.client.eql.EqlSearchResponse;
 import org.elasticsearch.client.eql.EqlStatsRequest;
 import org.elasticsearch.client.eql.EqlStatsResponse;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.time.DateUtils;
 import org.elasticsearch.index.IndexSettings;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.search.SearchHit;
 import org.junit.Before;
 
+import java.io.IOException;
 import java.time.format.DateTimeFormatter;
+import java.util.Locale;
+import java.util.Map;
 
+import static org.elasticsearch.common.xcontent.XContentFactory.jsonBuilder;
+import static org.hamcrest.Matchers.anyOf;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.greaterThan;
 
 public class EqlIT extends ESRestHighLevelClientTestCase {
 
+    private static final String INDEX_NAME = "index";
+    private static final int RECORD_COUNT = 40;
+    private static final int DIVIDER = 4;
+
     @Before
-    public void setupRemoteClusterConfig() throws Exception {
+    public void setup() throws Exception {
         setupRemoteClusterConfig("local_cluster");
+        setupData();
     }
 
-    public void testBasicSearch() throws Exception {
-        Request doc1 = new Request(HttpPut.METHOD_NAME, "/index/_doc/1");
-        doc1.setJsonEntity("{\"event_subtype_full\": \"already_running\", " +
-                "\"event\": {" +
-                    "\"category\": \"process\"" +
-                "}," +
-                "\"event_type_full\": \"process_event\", " +
-                "\"opcode\": 3," +
-                "\"pid\": 0," +
-                "\"process_name\": \"System Idle Process\"," +
-                "\"serial_event_id\": 1," +
-                "\"subtype\": \"create\"," +
-                "\"@timestamp\": 116444736000000000," +
-                "\"unique_pid\": 1}");
-        client().performRequest(doc1);
-        client().performRequest(new Request(HttpPost.METHOD_NAME, "/_refresh"));
+    private void setupData() throws IOException {
+        final BulkRequest bulkRequest = new BulkRequest();
+        bulkRequest.setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
+        for (int i = 0; i < RECORD_COUNT; i++) {
+            final IndexRequest indexRequest = new IndexRequest(INDEX_NAME);
+            indexRequest.source(jsonBuilder()
+                    .startObject()
+                    .field("event_subtype_full", "already_running")
+                    .startObject("event")
+                    .field("category", "process")
+                    .endObject()
+                    .field("event_type", "foo")
+                    .field("event_type_full", "process_event")
+                    .field("opcode", ((i % DIVIDER) == 0) ? 1 : 0)
```

Comment:
You could use the `randomXXX()` methods instead for arbitrary values or `randomBoolean()` to select which one should be used. This has the advantage of trying different combinations for all documents.